robotoff/app/core.py

```diff
@@ -107,5 +138,56 @@ def save_insight(
     if insight.annotation is not None:
         return ALREADY_ANNOTATED_RESULT
 
+    if verify_annotation:
+        verified: bool = False
```

Comment:
Opinion: that would be better to have a separate method for this part.

Reply:
I think that's a part of a reasonable cleanup with the transactions as well. A proper refactoring would probably split it out into one function that made votes and optionally also called a function that annotated insights if enough votes were collected - and the second function would either reuse or create a new transaction as needed.

Do we need it for this PR though?